inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -88,20 +88,27 @@ InferenceEngine::ICNNNetwork::Ptr clDNNEngine::CloneAndTransformNetwork(const In
             }
 
             // Reduce node implementation with reduce along features performs better with Reshape->Pooling->Reshape pattern
+            // Reshape->Pooling->Reshape scenario is also more optimal in case when batch = 16 and network precission is FP16
             if (auto redOp = std::dynamic_pointer_cast<const ::ngraph::opset1::ReduceMean>(node)) {
                 auto reduction_axes = redOp->get_reduction_axes().to_vector();
                 bool reduce_along_f = redOp->get_reduction_axes().size() == 1 && std::count(reduction_axes.begin(), reduction_axes.end(), 1) != 0;
-                return !reduce_along_f;
+                bool fp16_batch16 = redOp->get_element_type() == ngraph::element::f16 && redOp->input(0).get_shape()[0] == 16;
```

Comment:
fsv32 format can be selected for the batches in range  `[2; 16]`, so you need to enable this decomposition for the whole range.